antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/ParserRuleContext.kt

```diff
@@ -41,23 +41,33 @@ public open class ParserRuleContext : RuleContext {
     public val EMPTY: ParserRuleContext = ParserRuleContext()
   }
 
+  // This does not exist in the Java runtime
+  public val position: Position?
+    get() {
+      val start = this.start
+      val stop = this.stop
+
+      if (start != null && stop != null) {
```

Comment:
Perhaps we could write this as `if (start != null && stop?.endPoint() != null) {

Reply:
Nope, because you'd have to assign the result of `stop?.endPoint()` to a variable `x` anyway, to assert `x != null` to be able to pass it to `Position(start.startPoint(), x)`. So `stop?.endPoint() != null` is a redundant call to `endPoint()`.

Another argument for calling `startPoint()` and `endPoint()` only once is they're function, and the result may vary between invocations.